### PR TITLE
Regression reporting

### PR DESCRIPTION
**Pull request by amrasekh**
This adds regression model visualizations and reporting. 

An example case to try it out:

```
from sklearn.linear_model import LinearRegression
import pandas as pd
import credoai.lens as cl
from random import uniform

df = pd.DataFrame(
    [['Female', 1.1, 2], ['Female', 2, 3], ['Female', 2.2, 1], ['Female', 2.8, 2], 
    ['Male', 3, 4], ['Male', 1, 4], ['Male', 5.1, 3]],
    columns=['SEX', 'AGE', 'EDUCATION']
    )

df['target'] = df.apply(lambda x: (2 * x.EDUCATION + 3 * x.AGE) * uniform(0.9, 1.1), axis=1)

X = df[['AGE', 'EDUCATION']]
y = df['target']

model = LinearRegression().fit(X, y)

credo_model = cl.CredoModel(name='income_predictor',
                            model=model, 
                            model_config={'pred_fun': model.predict})
                            
credo_data = cl.CredoData(name='income_dataset',
                          data=df,
                          sensitive_feature_key='SEX',
                          label_key='target',
                          categorical_features_keys=['SEX', 'EDUCATION']
                          )

metrics = ['mean_absolute_error', 'explained_variance_score', 'r2_score', 'root_mean_squared_error']

alignment_spec = {'Fairness': {'metrics': metrics},
                  'Performance': {'metrics': metrics}}

lens = cl.Lens(model=credo_model,
               data=credo_data,
               spec=alignment_spec)
               
lens.run_assessments().display_results()
lens.create_report()
lens.get_report().write_report('reg_report.html')
```

### CODE REVIEW

File: credoai/reporting/model_fairness.py [hunk 1]
```diff
@@ -412,4 +411,111 @@ def _plot_text(self, f, text_objects):
...
+        df : pandas.dataframe
+            dataframe of true and predicted values, including also the associated demographic groups
+        label : str
+            plot title
+
+        Returns
+        -------
+        matplotlib figure
+        """        
+        y_true, y_pred = df['true'], df['pred']
```
Comment (IanAtCredo) [inline credoai/reporting/model_fairness.py:471]: If we have 10,000 datapoints, how does this plot fair? I'd probably place an upper limit on the number of data points by sampling without replacement.

**Review by IanAtCredo — CHANGES_REQUESTED** [review]
One question about how the scatter plot deals with large numbers of datapoints.

Could you also add a regression case to our test sets? I'd like to make sure this reporter is covered in our tests and make sure one metric (say r2) is correct.

File: credoai/reporting/model_fairness.py (hunk 1 above)
Comment (amrasekh) [reply credoai/reporting/model_fairness.py:476]: Good idea. This is now applied.

**Review by IanAtCredo — APPROVED** [review]
